docs/robot.md

```diff
@@ -0,0 +1,138 @@
+# Clusters with Robot Servers
+
+## Quickstart
+
+Prerequisites:
+
+- Running Cluster
+- Kubectl
+- Helm
+
+0. Make sure that you start all Kubelets in the cluster with `--cloud-provider=external`.
+
+1. Export credentials and create secret:
+
+   ```bash
+    export HCLOUD_TOKEN=your-hcloud-token
+    export ROBOT_USER=your-robot-user-name
+    export ROBOT_PASSWORD=your-robot-password
+   kubectl -n kube-system create secret generic hcloud --from-literal=token=$HCLOUD_TOKEN --from-literal=robot-user=$ROBOT_USER --from-literal=robot-password=$ROBOT_PASSWORD
+   ```
+
+2. Install the Helm Chart:
+
+   ```bash
+   helm repo add hcloud https://charts.hetzner.cloud
+   helm repo update hcloud
+   helm install hcloud/hcloud-cloud-controller-manager --set robot.enabled=true
+   ```
+
+You should now see that the Robot Server was initialized and the labels added.
+
+## Features
+
+Most of the features we support for Cloud servers are also supported for Robot servers:
+
+### Node Controller
+
+The Node controller adds information about the server to the Node object. The values are changed from what you usually see in the Robot interface & Webservice to better match the Cloud counterpart.
+
+- Labels
+  - `node.kubernetes.io/instance-type`
+    - Examples: `AX41` `Server-Auction`
+    - We replace any empty spaces with `-` (hyphen)
+  - `topology.kubernetes.io/region`
+    - Examples: `hel1` `fsn1`
+    - We use the lowercase variant of the location to match the Cloud Locations
+  - `topology.kubernetes.io/zone`
+    - Examples: `hel1-dc5` `fsn1-dc16`
+    - We use the lowercase variant of the location to match the Cloud Datacenters
+- Provider ID
+  - We set the field `Node.spec.providerID` to identify the Robot server after the initial adoption.
+  - The format is `hrobot://$SERVER_NUMBER`, but we can also read from the format used by [syself/hetzner-cloud-controller-manager](https://github.com/syself/hetzner-cloud-controller-manager): `hcloud://bm-$SERVER_NUMBER`
+- Addresses
+  - We add the Hostname and (depending on the configuration and availability) the IPv4 and IPv6 addresses of the server in `Node.status.addresses`.
+  - For the IPv6 address we use the first address in the Network -> For the network `2a01:f48:111:4221::` we add the address `2a01:f48:111:4221::1`.
+  - Private IPs in a vSwitch are not supported.
+
+### Node Lifecycle Controller
+
+The Node Lifecycle Controller is responsible for updating the shutdown status of Nodes & deleting the Kubernetes Node object if the corresponding server is removed.
+
+Both are generally supported. The shutdown status can only be detected if the Robot Server supports this.
```

Comment:
Do we have more details about which server handle this? I remember that this is about the age of the server?

Reply:
From the design doc: 

> Shutdown status is not available for server types using a tower case, this mostly affects older models

I did not find this information in the public Robot docs, nor did I find a list on which server models support it.